modules/vidoomyBidAdapter.js

```diff
@@ -0,0 +1,132 @@
+      [w, h] = bid.mediaTypes[BANNER].sizes[0];
+      adType = BANNER;
+    } else if (bid.mediaTypes && bid.mediaTypes[VIDEO] && bid.mediaTypes[VIDEO].playerSize) {
+      [w, h] = bid.mediaTypes[VIDEO].playerSize;
+      adType = VIDEO;
+    }
+
+    const host = bidderRequest.refererInfo.referer.split('#')[0].replace(/^(https\:\/\/|http\:\/\/)|(\/)$/g, '').split('/')[0];
+    const hostname = host.split(':')[0];
+
+    const videoContext = utils.deepAccess(bid, 'mediaTypes.video.context');
+
+    const queryParams = [];
+    queryParams.push(['id', bid.params.id]);
+    queryParams.push(['adtype', adType]);
+    queryParams.push(['w', w]);
+    queryParams.push(['h', h]);
+    queryParams.push(['pos', parseInt(bid.params.position) || 1]);
+    queryParams.push(['ua', navigator.userAgent]);
+    queryParams.push(['l', navigator.language && navigator.language.indexOf('-') !== -1 ? navigator.language.split('-')[0] : '']);
+    queryParams.push(['dt', /Mobi/.test(navigator.userAgent) ? 2 : 1]);
+    queryParams.push(['pid', bid.params.pid]);
+    queryParams.push(['dealId', bid.bidId]);
+    queryParams.push(['d', hostname]);
+    queryParams.push(['sp', encodeURIComponent(bidderRequest.refererInfo.referer)]);
+    if (bidderRequest.gdprConsent) {
+      queryParams.push(['gdpr', bidderRequest.gdprConsent.gdprApplies]);
+      queryParams.push(['gdprcs', bidderRequest.gdprConsent.consentString]);
+    }
+    queryParams.push(['usp', bidderRequest.uspConsent || '']);
+    queryParams.push(['coppa', !!config.getConfig('coppa')]);
+
+    const rawQueryParams = queryParams.map(qp => qp.join('=')).join('&');
+
+    const url = `${ENDPOINT}?${rawQueryParams}`;
+    return {
+      method: 'GET',
+      url: url,
+      data: {videoContext}
+    }
+  });
+  return serverRequests;
+};
+
+const render = (bid) => {
+  bid.ad = bid.vastUrl;
+  var obj = {
+    vastTimeout: 5000,
+    maxAllowedVastTagRedirects: 3,
+    allowVpaid: true,
+    autoPlay: true,
+    preload: true,
+    mute: true,
+  }
+  window.outstreamPlayer(bid, bid.adUnitCode, obj);
+}
+
+const interpretResponse = (serverResponse, bidRequest) => {
+  try {
+    let responseBody = serverResponse.body;
+    responseBody.requestId = responseBody.dealId;
+    if (responseBody.mediaType === 'video') {
+      responseBody.ad = responseBody.vastUrl;
+      const videoContext = bidRequest.data.videoContext;
+
+      if (videoContext === OUTSTREAM) {
+        try {
+          const renderer = Renderer.install({
+            id: bidRequest.bidId,
+            adunitcode: bidRequest.tagId,
+            loaded: false,
+            config: responseBody.mediaType,
+            url: responseBody.meta.rendererUrl
+          });
+          renderer.setRender(render);
+
+          responseBody.renderer = renderer;
+        } catch (e) {
+          responseBody.ad = responseBody.vastUrl;
+        }
+      }
+    }
+
+    return [responseBody];
```

Comment:
It would appear your serverResponse is essentially returning bids in the exact format needed for PBJS, however, it would be best if you built the object in your adapter to verify the existence of the properties and provide transparency to what is made available in your bidResponses. It's also hard to tell if you'll be passing back 1 or many here. Could you add that and look to other adapters to see examples.

Reply:
Ok, indeed, added explicitly

Reply:
Could we also do some validation confirming this is a good bid? IE) if it doesn't contain a **cpm** or an **ad** property it should not be used. Also, if it's a video request we should have **vastUrl** -- the `isBidResponseValid` function from lunamediahbBidAdapter is a good example of this.

Reply:
Ok, added the function, thanks for the example